base.go

```diff
@@ -366,16 +366,30 @@ func getBuilt(o BaseLayerOpts, sfm StackerFiles) error {
 		// it was sourced from e.g. a tar file, in which case there's
 		// nothing we can do besides initialize an empty oci tag and
 		// generate the whole thing.
+
+		// Attempt to copy the base from o.Config.OCIDir layout in case it's already there
+		err = lib.ImageCopy(lib.ImageCopyOpts{
+			Src:  fmt.Sprintf("oci:%s:%s", o.Config.OCIDir, baseInputTag),
+			Dest: fmt.Sprintf("oci:%s:%s", o.Config.OCIDir, targetName),
+		})
+		if err == nil {
+			return nil
+		}
+		fmt.Printf("warning: %s", err)
+
+		// Attempt to copy the base from the layer-bases layout
 		cacheDir := path.Join(o.Config.StackerDir, "layer-bases", "oci")
 		err = lib.ImageCopy(lib.ImageCopyOpts{
 			Src:  fmt.Sprintf("oci:%s:%s", cacheDir, baseInputTag),
 			Dest: fmt.Sprintf("oci:%s:%s", o.Config.OCIDir, targetName),
 		})
-		if err != nil {
-			// give up and just create an empty image
-			return umociInit(o)
+		if err == nil {
+			return nil
 		}
-		return nil
+		fmt.Printf("warning: %s", err)
+
+		// Give up and just create an empty image
+		return umociInit(o)
```

Comment:
If we drop the printf, I think we can drop this whole hunk, since it seems like a no-op?

Reply:
Is there a case for the image to be missing from both 'layer-bases' and 'o.Config.OCIDir'? If there is we still need this?

Reply:
Yes, I think so. For example, if you import a tar, for a build_only layer, but then base a layer that we do actually export on it later.

Reply:
In this case I don't get the question, what part can be dropped?

Reply:
This hunk just rearranges:
```
if err != nil {
    return umociInit(o)
}
```
to
```
if err == nil {
    return nil
}
return umociInit(o)
```
which I believe is a no-op.

Reply:
Reverted. I had rewritten it for consistency with the other copy. Where I wanted to avoid a nested if.